### DISCUSSION TITLE
Pyenv installs fine, but won't add a Python version?

### DISCUSSION THREAD

**Discussion by PsiMation** [pyenv/pyenv#2114] [General]
When trying to install a version of Python using pyenv, I get the following error below. I have gcc installed via Xcode. Can anyone point me in a direction to find the cause please?

pyenv install 3.9.0
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.9.0.tar.xz...
-> https://www.python.org/ftp/python/3.9.0/Python-3.9.0.tar.xz
Installing Python-3.9.0...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 11.6 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/_3/3n7g_jr918v2cj_pkdlfk5t40000gn/T/python-build.20211010191505.67494
Results logged to /var/folders/_3/3n7g_jr918v2cj_pkdlfk5t40000gn/T/python-build.20211010191505.67494.log

Last 10 log lines:
checking for python3... python3
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "darwin"
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in /var/folders/_3/3n7g_jr918v2cj_pkdlfk5t40000gn/T/python-build.20211010191505.67494/Python-3.9.0': configure: error: C compiler cannot create executables See config.log' for more details
make: *** No targets specified and no makefile found. Stop.

**Comment by hitblast** [comment]
It can either be the fact that you have an old version of **Command Line Tools for Xcode** installed, or you might not have accepted Xcode's license agreement. This can be fixed with the following command:


```bash
$ sudo gcc
You have not agreed to the Xcode license agreements. You must agree to both license agreements below in order to use Xcode.
[...]
```

If this doesn't fix the issue, then you might have to proceed with upgrading your Command Line Tools through the terminal or through **System Preferences > Software Update** respectively.

```bash
$ xcode-select --install
```

A similar issue can be noted on Mac OSX 10.8 in [this StackOverflow question](https://stackoverflow.com/questions/10357804/configure-error-c-compiler-cannot-create-executables) identical to what you've experienced.
